modules/logger/logger.go

```diff
@@ -6,31 +6,125 @@ import (
 	"io"
 	"os"
 	"runtime"
+	"strconv"
 	"strings"
+	gotesting "testing"
 	"time"
 
 	"github.com/gruntwork-io/terratest/modules/testing"
 )
 
-// Logf logs the given format and arguments, formatted using fmt.Sprintf, to stdout, along with a timestamp and information
-// about what test and file is doing the logging. This is an alternative to t.Logf that logs to stdout immediately,
-// rather than buffering all log output and only displaying it at the very end of the test. This is useful because:
-//
-// 1. It allows you to iterate faster locally, as you get feedback on whether your code changes are working as expected
-//    right away, rather than at the very end of the test run.
-//
-// 2. If you have a bug in your code that causes a test to never complete or if the test code crashes,, t.Logf would
-//    show you no log output whatsoever, making debugging very hard, where as this method will show you all the log
-//    output available.
-//
-// 3. If you have a test that takes a long time to complete, some CI systems will kill the test suite prematurely
-//    because there is no log output with t.Logf (e.g., CircleCI kills tests after 10 minutes of no log output). With
-//    this log method, you get log output continuously.
+// defaultLogf will be used if the caller uses the function
+// Logf, where on the first call to that function, a sane
+// default logging function will be set. We keep that in a
+// global variable so that we don't need to define it on
+// every call to Logf.
+var defaultLogf LogFunc
```

Comment:
Technically, we could make this variable public so that users can override it at run time.

This would mean that even if an "intermediate" package does not support setting a logger (for example, the k8s package currently), the user could set it himself.

However, it also means that packages could override these defaults and make a mess out of logging - that's why I kept it private.

What do you think?